src/gnu/prolog/database/PrologTextLoader.java

```diff
@@ -328,7 +328,15 @@ protected void processModuleDirective(Term moduleName, Term exports)
 			//PrologException.typeError(TermConstants.atomAtom, moduleName);
 			return;
 		}
-		prologTextLoaderState.startNewModule(this, (AtomTerm)moduleName, exports);
+		try
+		{
+			prologTextLoaderState.startNewModule(this, (AtomTerm)moduleName, exports);
+		}
+		catch (PrologException e)
+		{
+			e.printStackTrace();
+			logError("unable to start new module");
```

Comment:
I see what you mean about this use of logError rather than throwing exceptions. I can't quite remember why we do it this way? Is it so that all the compile errors get found at once rather than one at a time?